### PR TITLE
fix the v8 compiler to avoid some memory leaks for the generated binder code

### PR DESCRIPTION
**Pull request by ben-zhang-at-salesforce**
Primarily to address the memory leak in the generated binder code by the v8 compiler and did some cleanup...

### CODE REVIEW

**Comment by ben-zhang-at-salesforce** [comment]
memory fix 1: 
image

**Comment by ben-zhang-at-salesforce** [comment]
fix 2
image

**Comment by ben-zhang-at-salesforce** [comment]
2 more leaks 
image

**Review by jamie-houston — APPROVED** [review]

File: platforms/android/modules/compiler-v8/src/test/kotlin/com/salesforce/nimbus/bridge/v8/compiler/V8BinderGeneratorHelperTest.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import com.salesforce.nimbus.bridge.v8.compiler.V8BinderGeneratorHelper.CLASS_NAME_V8Function
+import com.salesforce.nimbus.bridge.v8.compiler.V8BinderGeneratorHelper.CLASS_NAME_V8Object
+import com.salesforce.nimbus.bridge.v8.compiler.V8BinderGeneratorHelper.CLASS_NAME_V8Releasable
+import com.squareup.kotlinpoet.MemberName.Companion.member
+import com.squareup.kotlinpoet.TypeSpec
+import com.squareup.kotlinpoet.asTypeVariableName
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import kotlin.reflect.full.memberProperties
```
Comment (threeve) [inline platforms/android/modules/compiler-v8/src/test/kotlin/com/salesforce/nimbus/bridge/v8/compiler/V8BinderGeneratorHelperTest.kt:23]: @ben-zhang-at-salesforce I'm getting `unresolved reference: full` on this line. Is there a dependency update that needs to be checked in along with this PR?

Comment (ben-zhang-at-salesforce) [reply platforms/android/modules/compiler-v8/src/test/kotlin/com/salesforce/nimbus/bridge/v8/compiler/V8BinderGeneratorHelperTest.kt]: removed the not finished test referring it.

**Pull request merged**